File: app/src/main/java/com/adiaz/deportesmadrid/db/entities/Favorite.java
package com.adiaz.deportesmadrid.db.entities;

import android.os.Parcelable;
import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class Favorite implements Parcelable {

    @Nullable
    public abstract Long id();

    @Nullable
    public abstract String idGroup();

    @Nullable
    public abstract Long idTeam();

    @Nullable
    public abstract String nameTeam();

    public static Favorite.Builder builder() {
        return new AutoValue_Favorite.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {
        public abstract Builder id(Long id);
        public abstract Builder idGroup(String idGroup);
        public abstract Builder idTeam(Long idTeam);
        public abstract Builder nameTeam(String nameTeam);
        public abstract Favorite build();
    }
}

File: app/src/main/java/com/adiaz/deportesmadrid/db/entities/Favorite.kt
package com.adiaz.deportesmadrid.db.entities

import android.os.Parcel
import android.os.Parcelable

class Favorite (val id: Long?=null, val idGroup: String, val idTeam: Long?=null, val nameTeam: String? = null) : Parcelable {

    constructor(parcel: Parcel) : this(
            parcel.readLong(),
            parcel.readString(),
            parcel.readLong(),
            parcel.readString()) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id!!)
        parcel.writeString(idGroup)
        parcel.writeLong(idTeam!!)
        parcel.writeString(nameTeam)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Favorite> {
        override fun createFromParcel(parcel: Parcel): Favorite {
            return Favorite(parcel)
        }

        override fun newArray(size: Int): Array<Favorite?> {
            return arrayOfNulls(size)
        }
    }

}
